step-release-vis/src/app/services/environment_test.ts

```diff
@@ -2,6 +2,13 @@ import { TestBed } from '@angular/core/testing';
 
 import { EnvironmentService } from './environment';
 import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { CandidateInfo } from '../models/Data';
+
+function fractionalEqual(x: number, y: number): boolean {
+  const relError: number = (x - y) / x;
+  return ((relError > -Math.pow(10, -7)) && (relError < Math.pow(10, -7)));
+}
+
 
```

Comment:
I think this is overengineering. We should use simpler data, where we do not need this and we can trust the built-in javascript division operator doing its job properly. (we don't need to test that)

Reply:
when we tried leaving the default equality comparator between numbers we got that 33.3 != 33.30000000004. so I came up with the relative error comparator as I know it is good practice and solves this problem

Reply:
That is correct, but what if we do not use 666,667,667 as job counts, but e.g 100,200,700. Do we still need this?

Reply:
What you need to test here is that the function will produce a map with correct percentages relative to the totalJobSum, not that the division operator work correctly.

Reply:
(make sure I understand) so basically should we get the percentages from the resultMap and see if they give us the same job count?